### PR TITLE
Topup API

### PR DESCRIPTION
**Pull request by tundeturner**
This adds the following stubs to Stripe ruby mock:

```
Stripe::Topup.create
Stripe::Topup.retreive
Stripe::Topup.list
```

### CODE REVIEW

File: lib/stripe_mock/instance.rb [hunk 1]
```diff
@@ -87,6 +88,7 @@ def initialize
       @country_spec = {}
       @tax_rates = {}
       @checkout_sessions = {}
+      @topups = Data.mock_disputes(['tu_05RsQX2eZvKYlo2C0FRTGSSA','tu_15RsQX2eZvKYlo2C0ERTYUIA', 'tu_25RsQX2eZvKYlo2C0ZXCVBNM', 'tu_35RsQX2eZvKYlo2C0QAZXSWE', 'tu_45RsQX2eZvKYlo2C0EDCVFRT', 'tu_55RsQX2eZvKYlo2C0OIKLJUY', 'tu_65RsQX2eZvKYlo2C0ASDFGHJ', 'tu_75RsQX2eZvKYlo2C0EDCXSWQ', 'tu_85RsQX2eZvKYlo2C0UJMCDET', 'tu_95RsQX2eZvKYlo2C0EDFRYUI'])
```
Comment (eyefodder) [inline lib/stripe_mock/instance.rb]: mock_disputes?

wouldnt you just want it to be `{}`

File: lib/stripe_mock/request_handlers/topups.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+      def assert_amount_valid(params)
+        if !params[:amount].to_i.positive?
+          raise Stripe::InvalidRequestError.new('Amount must be a positive integer', nil, http_status: 400)
+        end
+      end
+
+      def assert_currency_valid(params)
+        curr = params[:currency]
+        if curr.nil? || curr.match(/[[:lower:]]{3}/).nil?
```
Comment (eyefodder) [inline lib/stripe_mock/request_handlers/topups.rb]: @see `StripeMock::RequestHandlers::ParamValidatorsSUPPORTED_CURRENCIES.include?(params[:currency]`

looks like there;'s already currency handling in place

**Review by eyefodder — APPROVED** [review]
questionon default `topups` and currency validation

File: lib/stripe_mock/instance.rb (hunk 1 above)
Comment (tundeturner) [reply lib/stripe_mock/instance.rb]: @eyefodder Had a typo in a pattern I was following from the codebase.  Resolved now.

File: lib/stripe_mock/request_handlers/topups.rb (hunk 2 above)
Comment (tundeturner) [reply lib/stripe_mock/request_handlers/topups.rb]: thanks!

**Pull request merged**